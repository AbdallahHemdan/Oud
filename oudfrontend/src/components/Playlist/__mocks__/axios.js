export default {
    get: jest.fn(()=>{return Promise.resolve({
        "id": "2",
          "ownerName": "Jazz",
          "owner": "Ahmed Walid",
          "collaborative": true,
          "description": "Best music I have ever listened to",
          "followersCount": 100,
          "image":"https://img1.wsimg.com/isteam/ip/8203b8d1-fa5f-419d-a67f-e7d2656bf2ed/MEK_9025-1111.jpg",
          "public": true,
          "type": "trap",
          "name":"ahmed",
          "displayName":"ajmed",
          "tracks": [
            {
              "id": "21",
              "name": "إنسان",
              "artists": [
                {
                  "id": "5",
                  "name": "حمزة نمرة",
                  "type": "shaaby",
                  "image": "string"
                },
                {
                  "id": "2",
                  "name": "Sardeena",
                  "type": "Shaaby",
                  "image": "string"
                }

              ],
              "albumId": "1",
              "type": "Jazz",
              "audioUrl": "www.Facebook.com"
            },
            {
              "id": "22",
              "name": "سلاح التلميذ",
              "artists": [
                {
                  "id": "3",
                  "name": "ًWegz el wench",
                  "type": "Trap",
                  "image": "string"
                },
                {
                  "id": "4", 
                  "name": "Amr Diab",
                  "type": "farafeery",
                  "image": "string"
                }

              ],
              "albumId": "1",
              "type": "Trap",
              "audioUrl": "www.facebook.com"
            }
          ]
      })}),
      post: jest.fn(()=>{return Promise.resolve({status: "204", message: "success" })}),
      put: jest.fn(()=>{return Promise.resolve({status: "204", message: "success" })}),
      delete: jest.fn(()=>{return Promise.resolve({status: "204", message: "success" })})

};