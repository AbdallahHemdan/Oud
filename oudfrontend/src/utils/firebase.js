import firebase from "firebase";

const firebasConfig = {
  apiKey: "AIzaSyBZVUambAFEB-ywzAZTAPP-tSkbYXSEWc4",
  authDomain: "oud-notifications.firebaseapp.com",
  databaseURL: "https://oud-notifications.firebaseio.com",
  projectId: "oud-notifications",
  storageBucket: "oud-notifications.appspot.com",
  messagingSenderId: "681943816425",
  appId: "1:681943816425:web:9149a8834b2ff0f4623878"
};
// Initialize Firebase
firebase.initializeApp(firebasConfig);

export default firebase;
