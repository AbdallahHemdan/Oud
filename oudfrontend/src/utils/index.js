import axios from 'axios'
import { base } from '../config/environment'
const findByTestAttr = (component, val) => {
    return component.find(`[data-testid="${val}"]`);
}

function addToLikedSongs(id) {
    axios.put(`${base}/me/tracks/${id}`)
        .then(function (response) {
            console.log(response);

        })
        .catch(function (error) {
            console.log(error);
        });
}
function removeLikedSong(id) {
    axios.delete(`${base}/me/tracks/${id}`)
        .then(function (response) {
            console.log(response);
        })
        .catch(function (error) {
            console.log(error);
        });
}
export { findByTestAttr, addToLikedSongs, removeLikedSong }
