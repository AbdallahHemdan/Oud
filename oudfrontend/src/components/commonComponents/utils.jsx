import axios from "axios";
import { base } from "../../config/environment"
import {config} from "../../utils/auth"
/**
 * @file this file contains three functions used in Playlist, Album, LikedSongs 
 **/
export function addSong(playlistId, track){
  axios
  .post(`${base}/playlists/${playlistId}/${track}`, config)
  .then((response) => {
  })
  .catch((error) => {
  });
}