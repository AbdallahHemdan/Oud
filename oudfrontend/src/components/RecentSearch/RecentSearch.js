import React, { Component } from 'react'
import GenreCard from "./../GenreCard/GenreCard"
import RecentSearchCard from "./../RecentSearchCard/RecentSearchCard"
import LoadingSnipper from './../LoadingSnipper/LoadingSnipper';
import axios from "axios"
import { config } from "./../../utils/auth"
import { base, prodUrl } from "./../../config/environment"
import { Link } from 'react-router-dom';
// import "./RecentSearch.css"

/**
 * fetch url of all recent searches
 * @type {String}
 */
const fetchRecentSearchesUrl = (base === prodUrl) ? `${base}/me/search/recent` : `${base}/recentsearch`;

/**
 * Recent Search component
 *
 * @author Abdallah Hemdan
 *
 * @component
 *
 */
class RecentSearch extends Component {
  /**
   * @constructor
   *
   * @param {object} props - all recent search items
   * @param {string} items - list of items of recent search
   * @param {string} limit - the number of recent search items to return
   * @param {string} offset - The index of the first item in the recent search items
   * @param {boolean} total - The maximum number of items available to return
   * @param {string} isLoading - Determine if the data is already loaded in the component or not
   */
  constructor(props) {
    super(props)
    this.state = {
      /**
       * An array of all categories of the music to display it in home page
       *
       * @type {Array<object>}
       */
      items: [],
      /**
       * The maximum number of categories to get.
       *
       * @type {number}
       */
      limit: 0,
      /**
       * The index of the first categories to get.
       *
       * @type {number}
       */
      offset: 0,
      /**
       * The total number of categories available to get.
       *
       * @type {number}
       */
      total: 0,
      /**
      * Check if the data loaded from the backend or not
      * @type {Boolean}
      */
      isLoading: true
    }
  }
  /**
   * Function to handle storing data received from the props in the state
   * @function
   *
   * @return {void}
   *
   */
  handleStoringRecent = ({ items, limit, offset, total }) => {
    this.setState({ items, limit, offset, total });
  }

  /**
   * Fetching Current user recent search items
   *
   * @function
   * 
   * @returns {void} - nothing to return, it just fetch data and set it in the state
  */
  componentDidMount() {
    axios.get(fetchRecentSearchesUrl, config)
      .then((result) => {
        console.log("Data fetched on Recent search", result);
        this.handleStoringRecent(result.data);
      })
      .catch((err) => {
        console.log(err);
      });
  }

  componentDidUpdate(prevProps, preState) {
    if (preState.items !== this.state.items) {
      this.setState({ isLoading: false });
    }
  }
  /**
   * @function
   * @name render
   * @description Render all the recent search components
   *
   * @returns {JSX} Component for Home
  */
  render() {
    console.log("Recent Search State", this.state.items);
    return (
      <React.Fragment>
        {
          (this.state.isLoading && this.state.items.length) ?
            (
              <LoadingSnipper
                data-testid="loading"
              />
            )
            :
            (this.state.items.length) ?
              (
                <div className="module">
                  <div className="row"
                    data-testid="category-header"
                  >
                    <h1
                      className="gray-white item-name"
                      data-testid="category-title"
                    >Recent Search</h1>
                    {
                      (this.state.items.length >= 6) ?
                        <Link to={`/recent-search`}>
                          <div className="see-more"
                            data-testid="category-see-all"
                          >See All</div>
                        </Link> : null
                    }
                  </div>
                  <div
                    className="wrapper"
                    data-testid="first-wrapper">
                    <div className="wrapper_section_2"
                      data-testid="second-wrapper"
                    >
                      <div className="cards"
                        data-testid="cards-wrapper"
                      >
                        {
                          (this.state.items) ?
                            (this.state.items.splice(0, 6).map((item, index) => {
                              return (
                                <RecentSearchCard
                                  item={item}
                                  key={index}
                                  data-testid="recent-search-card"
                                />
                              )
                            })) : null
                        }
                      </div>
                    </div>
                  </div>
                </div>
              ) : null
        }
      </React.Fragment>
    )
  }
}

export default RecentSearch