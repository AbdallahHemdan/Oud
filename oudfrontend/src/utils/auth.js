import jwtDecode from "jwt-decode";
import { getRequest } from "./requester";
import { base } from "./../config/environment";

function getToken() {
  return localStorage.getItem("accessToken");
}

const config = {
  headers: { authorization: `Bearer ${getToken()}` }
};

function Auth() {
  try {
    if (!getToken()) return null;
    let decoded = jwtDecode(getToken());
    console.log(decoded.id);
    return decoded.id;
  } catch {
    return null;
  }
}
function isLoggedIn() {
  return getToken() ? true : false;
}

function isArtist() {
  return getRequest(`${base}/me`)
    .then(response => {
      return response.data.role === "artist";
    })
    .catch(error => {
      console.log(error);
    });
}
const _config = config;
export { _config as config };
export const token = getToken();
const _Auth = Auth;
export { _Auth as Auth };
const _isLoggedIn = isLoggedIn;
export { _isLoggedIn as isLoggedIn };
const _isArtist = isArtist;
export { _isArtist as isArtist };
