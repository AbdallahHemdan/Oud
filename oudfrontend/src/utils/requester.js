import axios from "axios";
import { isLoggedIn } from "./../utils/auth";
const config = {
  headers: {
    authorization: `Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjVlOTA3ZGIwYTA2NDVmNDU4MTYwNzYwNiIsImlhdCI6MTU5MTYyMTQxOSwiZXhwIjoxNTk0MjEzNDE5fQ.fj3N3Pc89Pf_xlt7fGmXw1SINTecUB4-y3pihAAPjC8`,
    "Access-Control-Allow-Origin": "*"
  }
};
const getRequest = endpoint => {
  return axios.get(endpoint, config);
};
/**
 * Axios DELETE request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @returns {object}
 */
const deleteRequest = endpoint => {
  return axios.delete(endpoint, config);
};
/**
 * Axios PUT request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const putRequest = (endpoint, body = {}) => {
  return axios.put(endpoint, body, config);
};
/**
 * Axios POST request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const postRequest = (endpoint, body = {}) => {
  return axios.post(endpoint, body, config);
};
/**
 * Axios PATCH request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const patchRequest = (endpoint, body = {}) => {
  return axios.patch(endpoint, body, config);
};

// const isArtist = () => {
//   if (isLoggedIn()) {
//   }
// };
export { getRequest, deleteRequest, putRequest, postRequest, patchRequest };
