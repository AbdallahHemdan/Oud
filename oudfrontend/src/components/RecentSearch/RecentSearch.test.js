import React from "react";
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer'
import { findByTestAttr } from "./../../utils/index"
import RecentSearch from './RecentSearch';
import checkPropTypes from 'check-prop-types'
import axios from "axios"

const setup = (props = {}) => {
  const component = shallow(
    <RecentSearch {...props} />
  )
  return component;
}

describe('Recent Search', () => {
  let component;
  beforeEach(() => {
    component = setup();
  })

  it('Should render category header in right way', () => {
    component.setState({ isLoading: false, items: [1, 2, 3, 4, 5, 6, 7, 8, 9] })
    const wrapper = findByTestAttr(component, "category-header");
    expect(component.state().isLoading).toBeFalsy();
    expect(wrapper.length).toBe(1);
  });

  it('Should render category header in right way', () => {
    component.setState({ isLoading: false, items: [1, 2, 3, 4, 5, 6, 7, 8, 9] })
    const wrapper = findByTestAttr(component, "category-title");
    expect(component.state().isLoading).toBeFalsy();
    expect(wrapper.length).toBe(1);
  });

  it('Should render recent search text in right way', () => {
    component.setState({ isLoading: false, items: [1, 2, 3, 4, 5, 6, 7, 8, 9] })
    const wrapper = findByTestAttr(component, "category-title");
    expect(component.state().isLoading).toBeFalsy();
    expect(wrapper.text()).toBe('Recent Search');
  });

  it('Should render category see all in right way', () => {
    component.setState({ isLoading: false, items: [1, 2, 3, 4, 5, 6, 7, 8, 9] })
    const wrapper = findByTestAttr(component, "category-see-all");
    expect(component.state().isLoading).toBeFalsy();
  });

  it('Should render first wrapper right way', () => {
    component.setState({ isLoading: false, items: [1, 2, 3, 4, 5, 6, 7, 8, 9] })
    const wrapper = findByTestAttr(component, "first-wrapper");
    expect(component.state().isLoading).toBeFalsy();
    expect(wrapper.length).toBe(1);
  });

  it('Should render second wrapper right way', () => {
    component.setState({ isLoading: false, items: [1, 2, 3, 4, 5, 6, 7, 8, 9] })
    const wrapper = findByTestAttr(component, "second-wrapper");
    expect(component.state().isLoading).toBeFalsy();
    expect(wrapper.length).toBe(1);
  });

  it('Should render Card wrapper in right way', () => {
    component.setState({ isLoading: false, items: [1, 2, 3, 4, 5, 6, 7, 8, 9] })
    const wrapper = findByTestAttr(component, "cards-wrapper");
    expect(component.state().isLoading).toBeFalsy();
    expect(wrapper.length).toBe(1);
  });

  it('Should render Recent search Cards in right way', () => {
    component.setState({ isLoading: false, items: [] })
    const wrapper = findByTestAttr(component, "Recent-search-card");
    expect(component.state().isLoading).toBeFalsy();
    expect(wrapper.length).toBe(component.state().items.splice(0, 6).length);
  });

  describe('snapshot test', () => {
    it('renders component correctly', () => {
      const tree = renderer
        .create(<RecentSearch />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

});