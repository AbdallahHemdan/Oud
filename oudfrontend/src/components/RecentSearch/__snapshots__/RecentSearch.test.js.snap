// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recent Search snapshot test renders component correctly 1`] = `null`;
